sessionctx/variable/session.go

```diff
@@ -956,6 +964,19 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 		SetMaxDeltaSchemaCount(tidbOptInt64(val, DefTiDBMaxDeltaSchemaCount))
 	case TiDBUsePlanBaselines:
 		s.UsePlanBaselines = TiDBOptOn(val)
+	case TiDBIsolationReadEngines:
+		s.isolationReadEngines = make([]kv.StoreType, 0, 2)
+		if len(val) == 0 {
+			break
+		}
+		for _, engine := range strings.Split(val, ",") {
+			switch engine {
+			case "TiFlash":
+				s.isolationReadEngines = append(s.isolationReadEngines, kv.TiFlash)
+			case "TiKV":
```

Comment:
What if there are duplicate engine names?

Reply:
I'll change the `isolationReadEngines` to map to avoid the situation.